src/components/internal/InternalInput.tsx

```diff
@@ -117,20 +125,23 @@ function InternalInput(props) {
     >
       <input
         css={inputCSS}
-        id={id}
         name={name}
+        ref={innerRef}
+        id={id}
         data-parent-name={parentName}
         placeholder={placeholder}
         type={type}
         {...variantProps}
-        maxLength={maxLength}
+        maxLength={
+          typeof maxLength === "string" ? parseInt(maxLength, 10) : maxLength
+        }
         disabled={disabled}
         onPaste={onPaste}
         autoComplete={autoComplete}
         autoCorrect="off"
         autoCapitalize="off"
         spellCheck="false"
-        aria-invalid={isValid ? null : "true"}
+        aria-invalid={isValid ? "false" : "true"}
         aria-describedby={describedBy}
```

Comment:
Can we not render `aria-invalid` when it's false and not render `aria-describedby` when it's undefined to make the markup a bit cleaner?

Reply:
I have updated `describedBy` to be undefined in `Input` component instead of defaulting to a string